### PR TITLE
commit for review

### PR DESCRIPTION
**Pull request by bramberry**

### CODE REVIEW

File: src/main/java/com/itechart/queryperfomance/queryperfomance/repository/SqlRepository.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    public SqlRepository(PropertiesConfig propertiesConfig) {
+        this.propertiesConfig = propertiesConfig;
+        init();
+    }
+
+    private void init() {
+        propertiesConfig.getInfo().forEach(info -> {
+            String url = String.format("%s:%s/%s", info.getUrl(), info.getPort(), info.getDbName());
+            try {
+                Connection connection = DriverManager
```
Comment (DenSev) [inline src/main/java/com/itechart/queryperfomance/queryperfomance/repository/SqlRepository.java]: Connection should be closed gracefully, use spring's @PreDestroy or DisposableBean. Additionally, using jdbc DataSource is preferred to using Connection.

File: src/main/java/com/itechart/queryperfomance/queryperfomance/servise/QueryService.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Autowired
+    public QueryService(SqlRepository repository) {
+        this.repository = repository;
+    }
+
+    public List<Answer> executeQuery(String query) {
+        Map<String, Connection> map = repository.getConnections();
+
+        ExecutorService service = Executors.newFixedThreadPool(10);
```
Comment (DenSev) [inline src/main/java/com/itechart/queryperfomance/queryperfomance/servise/QueryService.java]: The ExecutorService should be able to reuse threads, but in this code each request received by QueryService creates a new ExecutorService. The ExecutorService should be created in QueryService constructor. Additionally, ExecutorService should be shutdown at the end of application's lifecycle. Spring's @PreDestroy or DisposableBean could be used in this case.

Comment (DenSev) [reply src/main/java/com/itechart/queryperfomance/queryperfomance/servise/QueryService.java]: Also, this code does not ensure that only a single request is executed per single connection at any given point of time. It should be 1 Executor to 1 Connection.

File: src/main/java/com/itechart/queryperfomance/queryperfomance/repository/QueryExecutor.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    @Override
+    public Answer call() {
+        Answer answer = new Answer();
+        answer.setDbUrl(databaseUrl);
+        stopWatch.start();
+        try {
+            Statement st = connection.createStatement();
+            st.executeQuery(query);
+            stopWatch.stop();
+            answer.setTime(String.valueOf(stopWatch.getTotalTimeMillis()) + " " + TimeUnit.MILLISECONDS.toString());
```
[the remark this answers is not in the document]
Comment (bramberry) [reply src/main/java/com/itechart/queryperfomance/queryperfomance/repository/QueryExecutor.java]: fixed

File: src/main/java/com/itechart/queryperfomance/queryperfomance/repository/QueryExecutor.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        this.databaseUrl = databaseUrl;
+    }
+
+    @Override
+    public Answer call() {
+        Answer answer = new Answer();
+        answer.setDbUrl(databaseUrl);
+        stopWatch.start();
+        try {
+            Statement st = connection.createStatement();
```
[the remark this answers is not in the document]
Comment (bramberry) [reply src/main/java/com/itechart/queryperfomance/queryperfomance/repository/QueryExecutor.java]: fixed

File: src/main/java/com/itechart/queryperfomance/queryperfomance/repository/SqlRepository.java (hunk 1 above)
Comment (bramberry) [reply src/main/java/com/itechart/queryperfomance/queryperfomance/repository/SqlRepository.java]: fixed

File: src/main/java/com/itechart/queryperfomance/queryperfomance/repository/QueryExecutor.java [hunk 5]
```diff
@@ -27,15 +28,21 @@ public Answer call() {
         Answer answer = new Answer();
         answer.setDbUrl(databaseUrl);
         stopWatch.start();
+        Statement st = null;
         try {
-            Statement st = connection.createStatement();
```
Comment (DenSev) [inline src/main/java/com/itechart/queryperfomance/queryperfomance/repository/QueryExecutor.java]: try-with-resources is preferred way of handling AutoCloseable and Closeable classes. 

```java
try(Statement st = connection.createStatement()) {
```

File: src/main/java/com/itechart/queryperfomance/queryperfomance/repository/SqlRepository.java [hunk 6]
```diff
@@ -2,16 +2,19 @@
 
 
 import com.itechart.queryperfomance.queryperfomance.config.PropertiesConfig;
+import org.springframework.beans.factory.DisposableBean;
 import org.springframework.stereotype.Repository;
 
+import javax.annotation.PreDestroy;
```
Comment (DenSev) [inline src/main/java/com/itechart/queryperfomance/queryperfomance/repository/SqlRepository.java]: Make sure you have no unused imports.

File: src/main/java/com/itechart/queryperfomance/queryperfomance/servise/QueryService.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+        for (String name : map.keySet()) {
+            Callable<Answer> task = new QueryExecutor(name, map.get(name), query);
+            futures.add(executors.get(name).submit(task));
+        }
+        List<Answer> answers = new ArrayList<>();
+
+        futures.forEach(answerFuture -> {
+            try {
+                answers.add(answerFuture.get());
```
Comment (thehadalone) [inline src/main/java/com/itechart/queryperfomance/queryperfomance/servise/QueryService.java]: future.get() blocks the current thread, so we lost all benefits of using futures in this case. We need to wait until all futures are complete.

Comment (bramberry) [reply src/main/java/com/itechart/queryperfomance/queryperfomance/servise/QueryService.java]: I think, that I fixed it

Comment (bramberry) [reply src/main/java/com/itechart/queryperfomance/queryperfomance/servise/QueryService.java]: @thehadalone is it ok?

Comment (thehadalone) [reply src/main/java/com/itechart/queryperfomance/queryperfomance/servise/QueryService.java]: I think it's ok, no more objections.

**Pull request merged**